src/ValueFormatters/QuantityFormatter.php

```diff
@@ -168,4 +181,23 @@ private function formatMargin( DecimalValue $margin, $roundingExponent ) {
 		return null;
 	}
 
+	/**
+	 * @param string $number
+	 * @param string $unit
+	 *
+	 * @return string Text
+	 */
+	protected function applyUnit( $number, $unit ) {
+		if ( $this->vocabularyUriFormatter !== null ) {
+			$unit = $this->vocabularyUriFormatter->format( $unit );
+
+			if ( $unit === null ) {
+				return $number;
+			}
+		}
+
+		// TODO: localizable pattern for placement (before/after, separator)
+		return $number . ' ' . $unit;
```

Comment:
True. I think this is what we tried with the "unit appender interface" you called `QuantityUnitFormatter`. I would like to get rid of this and start over when we really need it.